iothub/device/samples/PnpDeviceSamples/TemperatureController/Parameter.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using CommandLine;
+
+namespace TemperatureController
+{
+    /// <summary>
+    /// Parameters for the application supplied via command line arguments.
+    /// If the parameter is not supplied via command line args, it will look for it in environment variables.
+    /// </summary>
+    internal class Parameters
+    {
+        [Option(
+            's',
+            "DeviceSecurityType",
+            HelpText = "(Required) The provisioning type that will be used for provisioning the device for the sample. Possible values include dps, connectionString (case-insensitive)." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_SECURITY_TYPE\".")]
+        public string DeviceSecurityType { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_SECURITY_TYPE");
+
+        [Option(
+            'p',
+            "PrimaryConnectionString",
+            HelpText = "(Required if DeviceSecurityType is \"connectionString\"). \nThe primary connection string for the device to simulate." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_CONNECTION_STRING\".")]
+        public string PrimaryConnectionString { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONNECTION_STRING");
+
+        [Option(
+            'e',
+            "DpsEndpoint",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe DPS endpoint to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_ENDPOINT\".")]
+        public string DpsEndpoint { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_ENDPOINT");
+
+        [Option(
+            'i',
+            "DpsIdScope",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe DPS ID Scope to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_ID_SCOPE\".")]
+        public string DpsIdScope { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_ID_SCOPE");
+
+        [Option(
+            'd',
+            "DeviceId",
```

Comment:
Do we really need DeviceId when we have DeviceConnectionString? Can we not extract it from the DeviceConnection string? It will lead to fewer user errors.

Reply:
They are required in different flows - the dps flow requires device Id, while the iothub flow requires the device connection string.

Reply:
Is the DeviceId going to be different from the one we can extract from the connection string? If not, can we just not extract it in our code so user does not have to have yet another input? And also if the deviceId in the connection string is not the same as the deviceId specified here, will the sample work?

Reply:
The input required for hub flow is device connection string.
The input required for dps flow is device Id, dps Id scope, dps endpoint, device symmetric key.

For dps flow, the user would need to supply the device Id, but not the device connection string. So there is no way for the sample to extract it from elsewhere.

```
  -s, --DeviceSecurityType         (Required) The provisioning type that will be used for provisioning the device for the sample. Possible values include dps,
                                                  connectionString (case-insensitive).
                                                  Defaults to environment variable "IOTHUB_DEVICE_SECURITY_TYPE".

  -p, --PrimaryConnectionString    **(Required if DeviceSecurityType is "connectionString").**
                                                      The primary connection string for the device to simulate.
                                                      Defaults to environment variable "IOTHUB_DEVICE_CONNECTION_STRING".

  -e, --DpsEndpoint                (Required if DeviceSecurityType is "dps").
                                               The DPS endpoint to use during device provisioning.
                                               Defaults to environment variable "IOTHUB_DEVICE_DPS_ENDPOINT".

  -i, --DpsIdScope                 (Required if DeviceSecurityType is "dps").
                                             The DPS ID Scope to use during device provisioning.
                                             Defaults to environment variable "IOTHUB_DEVICE_DPS_ID_SCOPE".

  -d, --DeviceId                   **(Required if DeviceSecurityType is "dps").**
                                           The device registration Id to use during device provisioning.
                                           Defaults to environment variable "IOTHUB_DEVICE_DPS_DEVICE_ID".

  -k, --DeviceSymmetricKey         (Required if DeviceSecurityType is "dps").
                                                    The device symmetric key to use during device provisioning.
                                                    Defaults to environment variable "IOTHUB_DEVICE_DPS_DEVICE_KEY".

```